model/identifier.py

```diff
@@ -859,6 +853,9 @@ def __eq__(self, other):
             return False
         return (self.type, self.identifier) == (other.type, other.identifier)
 
+    def __hash__(self):
+        return self.id
+
```

Comment:
I'm a little worried about the difference between `__eq__` (where two objects are the same if they're the same identifier) and `__hash__` (where two objects are the same if they're the same database row). Do you remember why you had to add this?

Reply:
Yeah, there are two tests that fail without this, `TestWork.test_reject_covers` and `TestEdition.test_equivalent_identifiers`. The error is `TypeError: unhashable type: 'Identifier'`. It looks like I can test for equivalency without putting the Identifiers in a `set(...)` but the `test_reject_covers` might need more than that to fix the test.

Reply:
what if you implemented hash as `hash((self.type, self.identifier))`?

Reply:
And, again, my suggestion is exactly what you did.